packages/app/components/DatePicker.tsx

```diff
@@ -0,0 +1,170 @@
+        <>
+          <Popover.Button
+            ref={setReferenceElement as Ref<HTMLButtonElement>}
+            className={twMerge(
+              "flex justify-between items-center focus:border-0",
+              className
+            )} // text-em-low if default dates
+          >
+            <span>{formattedDate()}</span>
+            <Icon name="caret-down" className="mr-2 h-4 w-4 text-black" />
+          </Popover.Button>
+          {open && (
+            <>
+              <div className="fixed inset-0 bg-black/30" aria-hidden="true" />
+              <Popover.Panel
+                ref={setPopperElement as Ref<HTMLDivElement>}
+                style={styles.popper}
+                {...attributes.popper}
+                className="w-fit h-fit m-auto max-md:!fixed max-md:!inset-0 max-md:!flex max-md:!items-center max-md:!justify-center max-md:!transform" /* This styles will make it hover on mobile */
+              >
+                {({ close }) => (
+                  <div className="flex flex-col space-y-2 bg-white divide-y divide-surface-50 rounded-2xl border border-surface-50">
+                    <Calendar
+                      mode="single"
+                      selected={currentDate}
+                      defaultMonth={currentDate}
+                      onSelect={(newDate: Date | undefined) => {
+                        if (!newDate) return;
+                        newDate.setHours(currentDate.getHours());
+                        newDate.setMinutes(currentDate.getMinutes());
+                        setCurrentDate(newDate);
+                      }}
+                      initialFocus
+                      className=""
+                      fromDate={fromDate || new Date()}
+                    />
+                    <div className="flex flex-col space-y-2 p-4 pt-2">
+                      <div className="flex justify-between items-center">
+                        <span>{timeCaption}</span>
+                        <div className="flex space-x-2  items-center">
+                          <div className="flex items-center border border-surface-75 rounded-xl py-2 px-3 text-em-low">
+                            <input
+                              className="outline-none font-semibold flex-grow text-sm w-5 text-center"
+                              type="number"
+                              pattern="[0-9]*"
+                              value={hours}
+                              onKeyDown={(evt) =>
+                                ["e", "E", "+", "-"].includes(evt.key) &&
+                                evt.preventDefault()
+                              }
+                              onChange={(event) => {
+                                const value = event.target.value;
+                                const numberValue = Number(value);
+                                if (
+                                  numberValue >= 0 &&
+                                  numberValue < 24 &&
```

Comment:
```suggestion
                                const hours = Number(value);
                                if (
                                  hours >= 0 &&
                                  hours < 24 &&
```

`numberOfHours` would also work